File: 20.Valid_Parentheses.cpp
class Solution {
    public:
    bool isValid(string s) {
        vector<char> res;
        for (int i=0; i<s.length(); i++){
            if (s[i] == '('){
                res.push_back(')');
            }
            else if (s[i] == '['){
                res.push_back(']');
            }
            else if (s[i] == '{'){
                res.push_back('}');
            }
            else {
                if (res.empty() || res.back() != s[i]) return false;
                res.pop_back();
            }
        }
        return res.empty();
    }
};

File: 20.ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        dic = {'(':')', '[': ']', '{': '}'}
        for c in s:
            if c in dic.keys():
                stack.append(dic[c])
            else:
                if len(stack) == 0 or c != stack.pop():
                    return False
        return True if len(stack) == 0 else False